test/Microsoft.AspNetCore.Routing.Tests/LinkGeneratorRouteValuesAddressExtensionsTest.cs

```diff
@@ -17,6 +17,64 @@ namespace Microsoft.AspNetCore.Routing
     // Does not cover the RouteValuesAddressScheme in detail. see RouteValuesAddressSchemeTest
     public class LinkGeneratorRouteValuesAddressExtensionsTest : LinkGeneratorTestBase
     {
+        [Fact]
+        public void GetPathByRouteValues_UsePrioritizedEndpointOnlyWithMatchingRouteValue()
```

Comment:
This could really be called `UsesFirstTemplateThatSucceeds`.

Reply:
This is really a test for https://github.com/aspnet/Routing/blob/release/2.2/test/Microsoft.AspNetCore.Routing.Tests/DefaultLinkGeneratorTest.cs - it's super general and isn't related to these overloads.